src/app/pages/product/product-details/product-details.component.html

```diff
@@ -1,35 +1,115 @@
+            <li class="product__item">
+                <span class="product__item__label">Size:</span>
+                <span>{{product.sizes | titlecase}}</span>
+            </li>
+            <li class="product__item">
+                <span class="product__item__label">Buy by:</span>
+                <span>{{product.buyProperties}}</span>
+            </li>
+            <li class="product__item">
+                <span class="product__item__label">Delivery:</span>
+                <span>in {{product.deliveryTime}} days</span>
+            </li>
+            <li class="product__item">
+                <span class="product__item__label">Delivery area:</span>
+                <span>{{product.deliveryArea | titlecase}}</span>
+            </li>
+        </ul>
+        <div class="product__actions">
+            <div class="product__price">
+                <span class="product__price-current">{{product.price}} USD</span>
+                <span class="product__price-previous">{{product.previousPrice}} USD</span>
+            </div>
+            <form [formGroup]="detailsForm" class="product__amount">
+                <div class="product__amount-number">
+                    <input type="number" min="1" formControlName="count" (change)="handleChangeValue($event)">
+                    <label>{{this.detailsForm.get('measurementUnit').value | lowercase}}</label>
+                </div>
+                <hr>
+                <select class="select" formControlName="measurementUnit">
+                    <option selected>Psc</option>
+                    <option>Sdc</option>
+                </select>
+            </form>
+            <button class="green-button" (click)="handleAddToCart(product)">
+                <img src="assets/images/svg/ic-actions-add-simple.svg" alt="add">
+                <a>Add to cart</a>
+            </button>
         </div>
-        <p>Add to my wish list</p>
-        <div>
-            We work hard to ensure that the fruit and vegetables we sell are fresh and high in quality. If we don’t grow
-            them
-            ourselves, we source them from carefully chosen suppliers, preferring to buy locally whenever possible.
+        <button class="wish-button">
+            <img src="assets/images/svg/ic-heart.svg">
+            <a>Add to my wish list</a>
+        </button>
+        <div class="product__tabs">
+            <app-tabs>
+                <app-tab title="Description">
+                    <div class="tab__block">
+                        <h3 class="tab__title">Origins</h3>
+                        <p class="tab__content">{{product.description.origins}}</p>
+                    </div>
+                    <div class="tab__block">
+                        <h3 class="tab__title">How to cook</h3>
+                        <p class="tab__content">{{product.description.howToCook}}</p>
+                    </div>
+                </app-tab>
+                <app-tab title="Reviews" [count]="product.reviews.length">
```

Comment:
```suggestion
                <app-tab title="Reviews" [count]="product.reviews?.length">
```
If there aren't any reviews you will get an error in the way you wrote